py/trtorch/ptq.py

```diff
@@ -0,0 +1,61 @@
+from typing import List, Dict, Any
+import torch
+import os
+
+import trtorch._C
+from trtorch._compile_spec import _parse_compile_spec
+from trtorch._version import __version__
+from types import FunctionType
+import tensorrt as trt
+
+def get_cache_mode_batch(self):
+    return None
+
+def get_batch_size(self):
+    return self.batch_size
+
+def get_batch(self, names):
+    if self.current_batch_idx + self.batch_size > self.data_loader.dataset.data.shape[0]:
+        return None
+    batch = self.dataset_iterator.next()
+    self.current_batch_idx += self.batch_size
+    print("Calibrating batch: ", self.current_batch_idx)
+    # Treat the first element as input and others as targets.
+    if isinstance(batch, list):
+        batch = batch[0].to(torch.device('cuda:0'))
+    return [batch.data_ptr()]
+
+def read_calibration_cache(self):
+    if self.use_cache:
+        if os.path.exists(self.cache_file):
+            with open(self.cache_file, "rb") as f:
+                return f.read()
+
+def write_calibration_cache(self, cache):
+    with open(self.cache_file, "wb") as f:
+        f.write(cache)
+
+def make_int8_calibrator(dataloader, cache_file, use_cache, algo_type):

```

Comment:
Lets use kwargs for everything except the dataloader

Reply:
Replaced this with a class